### ISSUE TITLE
The compatibility with SpaceRanger v2.0 or later of `read_visium` function

### ISSUE THREAD

**Issue by senbaikang** [scverse/scanpy#2746]
### Please make sure these conditions are met

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the latest version of scanpy.
- [X] (optional) I have confirmed this bug exists on the master branch of scanpy.

### What happened?

When reading the `tissue_positions.csv` file generated by SpaceRanger v2.0 or later, `read_visium` reads the header from the second row (with `header=1`), while it should read from the first row instead (with `header=0`).

### Minimal code sample

```python
positions = pd.read_csv(
            files['tissue_positions_file'],
            header=0 if tissue_positions_file.name == "tissue_positions.csv" else None,
            index_col=0,
        )
```

### Versions

<details>

```
anndata     0.10.2
scanpy      1.10.0.dev157+g9b7f6032
```

</details>

**Comment by senbaikang** [comment]
This has also been mentioned by @flying-sheep [here](https://github.com/scverse/scanpy/pull/2424#issuecomment-1607082327).

**Comment by senbaikang** [comment]
The fix: #2743
